src/main/scala/NIC.scala

```diff
@@ -211,8 +210,14 @@ class IceNicSendPath(nInputTaps: Int = 0)(implicit p: Parameters)
     } else { reader.module.io.out }
 
     val unlimitedOut = if (nInputTaps > 0) {
-      val arb = Module(new PacketArbiter(1 + nInputTaps, rr = true))
-      arb.io.in <> (preArbOut +: io.tap.get)
+      val bufWords = (packetMaxBytes - 1) / NET_IF_BYTES + 1
+      val inputs = (preArbOut +: io.tap).map { in =>
```

Comment:
Why do you need to buffer the packets before passing to the arbiter? Are packets potentially dropped in the old approach?

Reply:
Nevermind. Didn't read the PR description.